src/sink/elasticsearch.rs

```diff
@@ -169,24 +185,40 @@ impl Sink for Elasticsearch {
 
         let mut buffer = String::with_capacity(4048);
         self.bulk_body(&mut buffer);
-        debug!("BODY: {:?}", buffer);
         let bulk_resp: Result<BulkResponse> = client
             .request(BulkRequest::new(buffer))
             .send()
             .unwrap()
             .into_response::<BulkResponse>();
 
+        ELASTIC_INTERNAL_BUFFER_LEN.store(self.buffer.len(), Ordering::Relaxed);
         match bulk_resp {
             Ok(bulk) => {
-                self.buffer.clear();
                 ELASTIC_RECORDS_DELIVERY.fetch_add(1, Ordering::Relaxed);
                 for item in bulk.iter() {
                     match item {
-                        Ok(_item) => {
+                        Ok(item) => {
+                            let uuid = uuid::Uuid::parse_str(item.id())
+                                .expect("catastrophic error, TID not a UUID");
+                            if let Ok(idx) = self.buffer
+                                .binary_search_by(|probe| probe.uuid.cmp(&uuid))
+                            {
+                                self.buffer.remove(idx);
+                            }
                             ELASTIC_RECORDS_TOTAL_DELIVERED
                                 .fetch_add(1, Ordering::Relaxed);
                         }
                         Err(item) => {
+                            let uuid = uuid::Uuid::parse_str(item.id())
```

Comment:
What not parse this uuid just before this match block?

Reply:
The type doesn't work out. Before this block the `item` is a `Result<Item, ?>` and here it's the item we need.